### PR TITLE
Add attributedText binding target to UITextView and UITextField

### PR DESCRIPTION
**Pull request by augusteo**
very simple attributedText binding target for `UITextView` and `UITextField`.

### CODE REVIEW

File: ReactiveCocoa/UIKit/UITextView.swift [hunk 1]
```diff
@@ -7,6 +7,11 @@ extension Reactive where Base: UITextView {
 	public var text: BindingTarget<String> {
```
Comment (andersio) [inline ReactiveCocoa/UIKit/UITextView.swift]: Makes me wonder if this should be `String?` instead, or instead we hide the nullability and consistently use an empty string in our APIs.

**Comment by andersio** [comment]
```
❌  /Users/travis/build/ReactiveCocoa/ReactiveCocoa/ReactiveCocoaTests/UIKit/UITextViewSpec.swift:32:62: use of undeclared type 'NoError'
                                let (pipeSignal, observer) = Signal<NSAttributedString?, NoError>.pipe()
                                                                                         ^~~~~~~
```
The UITextViewSpec needs to import `enum Result.NoError`.

File: ReactiveCocoa/UIKit/UITextView.swift (hunk 1 above)
Comment (augusteo) [inline ReactiveCocoa/UIKit/UITextView.swift]: `UITextView` text property is optional, so yes, `String?` would be closer to the truth, the same with label and textfield. Do you want me to make the change?

**Comment by augusteo** [comment]
I've imported `Result.NoError`

File: ReactiveCocoa/UIKit/UITextView.swift (hunk 1 above)
Comment (andersio) [inline ReactiveCocoa/UIKit/UITextView.swift]: It is somehow IUO though. But yeah, please turn them into `String?`. Thanks.

**Comment by andersio** [comment]
May you also add the signals for `NSAttributedString`?

**Comment by augusteo** [comment]
> May you also add the signals for NSAttributedString?

what do you mean by that @andersio

**Comment by andersio** [comment]
`attributedTextValues` and `continuousAttributedTextValues`.

**Comment by augusteo** [comment]
Ah got it. will do

File: ReactiveCocoaTests/UIKit/UITextViewSpec.swift [hunk 2]
```diff
@@ -3,50 +3,91 @@ import ReactiveCocoa
...
-				}
+		it("should emit user initiated changes to its text value when the editing ends") {
+			textView.text = "Test"
 
-				NotificationCenter.default.post(name: .UITextViewTextDidEndEditing,
-				object: textView)
-				expect(latestValue) == textView.text
+			var latestValue: String?
+			textView.reactive.textValues.observeValues { text in
+			latestValue = text
```
Comment (andersio) [inline ReactiveCocoaTests/UIKit/UITextViewSpec.swift]: Please fix the Indentation. 😸

File: ReactiveCocoaTests/UIKit/UITextViewSpec.swift [hunk 3]
```diff
@@ -3,50 +3,91 @@ import ReactiveCocoa
...
 			}
 
-			it("should emit user initiated changes to its text value continuously") {
-				textView.text = "Test"
+			NotificationCenter.default.post(name: .UITextViewTextDidEndEditing,
+			object: textView)
```
Comment (andersio) [inline ReactiveCocoaTests/UIKit/UITextViewSpec.swift]: Ditto.

File: ReactiveCocoaTests/UIKit/UITextViewSpec.swift [hunk 4]
```diff
@@ -3,50 +3,91 @@ import ReactiveCocoa
...
+			textView.text = "Test"
 
-				var latestValue: String?
-					textView.reactive.continuousTextValues.observeValues { text in
-						latestValue = text
-				}
+			var latestValue: String?
+				textView.reactive.continuousTextValues.observeValues { text in
+					latestValue = text
+			}
```
Comment (andersio) [inline ReactiveCocoaTests/UIKit/UITextViewSpec.swift]: Ditto.

File: ReactiveCocoaTests/UIKit/UITextViewSpec.swift [hunk 5]
```diff
@@ -3,50 +3,91 @@ import ReactiveCocoa
...
 
-				NotificationCenter.default.post(name: .UITextViewTextDidChange,
-				object: textView)
-				expect(latestValue) == textView.text
+			NotificationCenter.default.post(name: .UITextViewTextDidChange,
+			object: textView)
```
Comment (andersio) [inline ReactiveCocoaTests/UIKit/UITextViewSpec.swift]: Ditto.

File: ReactiveCocoa/UIKit/UITextView.swift [hunk 6]
```diff
@@ -30,4 +30,32 @@ extension Reactive where Base: UITextView {
 	public var continuousTextValues: Signal<String, NoError> {
```
Comment (andersio) [inline ReactiveCocoa/UIKit/UITextView.swift]: Can you turn `textValues` and `continuousTextValues` to use `String?` too?

**Comment by andersio** [comment]
Thanks! It should be ready to go after these changes. :shipit:

**Comment by augusteo** [comment]
should be all good. thanks for the feedback @andersio

**Comment by andersio** [comment]
Gonna merge after CI test passes.

File: ReactiveCocoa/UIKit/UITextView.swift [hunk 7]
```diff
@@ -20,14 +20,42 @@ extension Reactive where Base: UITextView {
...
+	public var continuousTextValues: Signal<String?, NoError> {
 		return textValues(forName: .UITextViewTextDidChange)
 	}
+	
+	/// Sets the attributed text of the text view.
+	public var attributedText: BindingTarget<NSAttributedString?> {
+		return makeBindingTarget { $0.attributedText = $1 }
+	}
+	
+	private func attributedTextValues(forName name: NSNotification.Name) -> Signal<String, NoError> {
```
Comment (andersio) [inline ReactiveCocoa/UIKit/UITextView.swift]: Oops. The return type is incorrect.

**Comment by augusteo** [comment]
sorry, should be fine now

**Comment by augusteo** [comment]
i found some test issue where the textview styling is changing the attributed text styling and fails the test. I'll try to fix it.

**Comment by augusteo** [comment]
ok the tests are all passing now.

I had to make some changes.
in `UITextViewSpec`:
added attributes in the test expectation because without it, the `textView` would override the styles.

in `UITextFieldSpec`:
adding the two attribute as per `textView` doesn't work because the  `textField` added a lot more attributes, failing the test. So I just tested the `attributedText.string` which is quite static, and still shows that the method will send the correct string when triggered

**Comment by augusteo** [comment]
CI server errors, should be unrelated to the actual code change:

```
2016-11-24 22:44:10.608 xcodebuild[1849:6394] Error Domain=IDETestOperationsObserverErrorDomain Code=5 "Early unexpected exit, operation never finished bootstrapping - no restart will be attempted" UserInfo={NSLocalizedDescription=Early unexpected exit, operation never finished bootstrapping - no restart will be attempted}
** TEST EXECUTE FAILED **

*** Building scheme "Nimble-macOS" in Nimble.xcodeproj
xcodebuild timed out while trying to read ReactiveCocoa.xcworkspace 😭
The command "carthage build --no-skip-current --platform mac" exited with 1.
```

**Comment by andersio** [comment]
@augusteo Could you try running the tvOS tests locally?

**Comment by augusteo** [comment]
Ok turns out tvOS doesn't have `Georgia` font, that's why its breaking on tvOS. I've just replaced it with `systemFont(:)` which works on both iOS and tvOS now

**Comment by augusteo** [comment]
yay all green!

**Pull request merged**